gcp_variant_transforms/pipeline_common.py

```diff
@@ -101,6 +104,40 @@ def _get_all_patterns(input_pattern, input_file):
   return patterns
 
 
+def get_compression_type(input_patterns):
+  # type: (List[str]) -> filesystem.CompressionTypes
+  """Returns the compression type.
+
+  Raises:
+    ValueError: if the input files are not in the same format.
+  """
+  matches = filesystems.FileSystems.match(input_patterns)
+  extensions = [os.path.splitext(metadata.path)[1] for match in matches
+                for metadata in match.metadata_list]
+  if len(set(extensions)) != 1:
+    raise ValueError('All input files must be in the same format.')
+  if extensions[0].endswith('.bgz') or extensions[0].endswith('.gz'):
+    return filesystem.CompressionTypes.GZIP
+  else:
+    return filesystem.CompressionTypes.AUTO
+
+
+def get_splittable_bgzf(all_patterns):
+  # type: (List[str]) -> List[str]
+  """Returns the splittable bgzf matching `all_patterns`."""
+  matches = filesystems.FileSystems.match(all_patterns)
+  files = []
+  for match in matches:
+    for metadata in match.metadata_list:
+      if not (metadata.path.startswith('gs://') and
+              bgzf_io.exists_tbi_file(metadata.path)):
+        logging.info("Cannot find the index file for %s.", metadata.path)
+        return []
```

Comment:
If the intent is to discard all the previous matches, shouldn't this be an error and raise an exception?